rig/routing_table/utils.py

```diff
@@ -1,6 +1,55 @@
 import warnings
 
-from rig.routing_table import RoutingTableEntry
+from rig.routing_table import RoutingTableEntry, MinimisationFailedError
+
+
+def remove_default_routes(table, target_length):
```

Comment:
I would prefer to present this as a routing-table minimisation algorithm. This would mean putting it in a module named after the algorithm and calling it "minimise". Would you be OK with that? (It already meets other API requirements, I think).